tox.ini

```diff
@@ -14,7 +14,7 @@ setenv =
   QISKIT_SUPRESS_PACKAGING_WARNINGS=Y
   QISKIT_TEST_CAPTURE_STREAMS=1
   QISKIT_PARALLEL=FALSE
-passenv = RAYON_NUM_THREADS OMP_NUM_THREADS QISKIT_PARALLEL RUST_BACKTRACE SETUPTOOLS_ENABLE_FEATURES
+passenv = RAYON_NUM_THREADS OMP_NUM_THREADS QISKIT_PARALLEL RUST_BACKTRACE SETUPTOOLS_ENABLE_FEATURES RUSTFLAGS CARGO_INCREMENTAL LLVM_PROFILE_FILE
```

Comment:
Do we still need these changes if we're not using `tox -ecoverage`?

Reply:
I left them in for local testing. It makes it convenient to set these flags if you want to collect coverage locally. Although I guess it would be a better argument to just set these as hard coded env variables in the coverage job itself.

Reply:
I'm vaguely concerned that `RUSTFLAGS` could easily be in somebody's environment which could cause their `tox` to have different behaviour to our CI, but honestly, the chance of that happening (and the impact if it does) is so minimal that if it's helpful to you for testing, let's just leave it.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/8365/commits/630d99990db73cc2a9f26e5610e0d18e318c37e2